substrate/frame/revive/src/limits.rs

```diff
@@ -91,22 +91,22 @@ pub mod code {
 	/// the allowed code size by [`BYTE_PER_INSTRUCTION`].
 	pub const STATIC_MEMORY_BYTES: u32 = 2 * 1024 * 1024;
 
+	/// The maximum size of a basic block in number of instructions.
+	///
+	/// We need to limit the size of basic blocks because the interpreters lazy compilation
+	/// compiles one basic block at a time. A malicious program could trigger the compilation
+	/// of the whole program by creating one giant basic block otherwise.
+	pub const BASIC_BLOCK_SIZE: u32 = 1000;
```

Comment:
Can we safely bump this number? Since we've observed even basic Solidity contracts exceed this without the PVM optimizer enabled, I feel like it could be worth to bump it further if possible to give some more margin.

Reply:
Dumb Question: Could this be a `Config` variable so we could change that in the future?

Reply:
> Dumb Question: Could this be a Config variable so we could change that in the future?
I deliberately removed everything from the `Config` and hardcoded it here for those reasons:
1) Make configuration of the pallet easier
2) To not fragment the ecosystem in case the usage of the pallet becomes pervasive. The same reason we hardcoded all the types. A contract should be able to run everywhere.

> Can we safely bump this number? 
Yes! We can bump all numbers in this file without any issue (but never reduce). In order to make that possible I needed to add static validation (in `limits::enforce`) that is only run on code upload but never applied to already existing code.

Reply:
> I deliberately removed everything from the Config and hardcoded it here for those reasons:

Yes you are right. That is a lesson we learned the hard way from the contracts pallet days.

> Yes! All numbers in the code module can be changed without any issue.

Great, thanks for the clarification!